src/org/zaproxy/zap/spider/parser/SpiderParser.java

```diff
@@ -93,10 +95,12 @@ protected void notifyListenersPostResourceFound(HttpMessage message, int depth,
 	 * @param depth the depth
 	 * @param localURL the local url
 	 * @param baseURL the base url
+	 * @param irrelevantUrlParameters url parameters that are skipped when canonicalizing
 	 */
-	protected void processURL(HttpMessage message, int depth, String localURL, String baseURL) {
+	protected void processURL(HttpMessage message, int depth, String localURL, String baseURL,
+							  Set<String> irrelevantUrlParameters) {
```

Comment:
Since all parsers end up passing the same irrelevant parameters I'd suggest adding a new constructor `SpiderParser(SpiderParam)` (add deprecate the default) instead of adding this.